Mage/src/main/java/mage/game/permanent/PermanentImpl.java

```diff
@@ -1053,7 +1053,7 @@ private int doDamage(int damageAmount, UUID attackerId, Ability source, Game gam
             if (attacker != null && markDamage) {
```

Comment:
There are two potential places to check -- make sure new event logic compatible with it (it's also a good things for unit tests/combo check):
1. markDamage (combined combat damage from multiple sources to planeswalker/battle);
2. checkWither/M1M1 (damage from Infect and Wither abilities).

Reply:
on point 2: wither and such dont remove counters, so what needs to be tested in regards to them?